File: io-hotmoka-network/src/main/java/io/hotmoka/network/models/values/StorageReferenceModel.java
package io.hotmoka.network.models.values;

import java.math.BigInteger;

import io.hotmoka.beans.annotations.Immutable;
import io.hotmoka.beans.values.StorageReference;

@Immutable
public class StorageReferenceModel {
	public final TransactionReferenceModel transaction;
    public final String progressive;

    public StorageReferenceModel(StorageReference input) {
    	transaction = new TransactionReferenceModel(input.transaction);
    	progressive = input.progressive.toString();
    }

    public StorageReference toBean() {
    	return new StorageReference(transaction.toBean(), new BigInteger(progressive));
    }
}

File: io-hotmoka-network-thin-client/io-hotmoka-network-thin-client-kt/src/main/kotlin/io/hotmoka/network/thin/client/models/values/StorageReferenceModel.kt
package io.hotmoka.network.thin.client.models.values

class StorageReferenceModel(val transaction: TransactionReferenceModel, val progressive: String) {


    override fun equals(other: Any?): Boolean {
        return other is StorageReferenceModel &&
                other.transaction == transaction && other.progressive == progressive
    }

    override fun hashCode(): Int {
        return progressive.hashCode() xor transaction.hashCode()
    }

}